sdk/src/main/java/com/kin/ecosystem/settings/presenter/SettingsPresenter.kt

```diff
@@ -0,0 +1,182 @@
+package com.kin.ecosystem.settings.presenter
+
+import android.content.Intent
+import com.kin.ecosystem.base.BasePresenter
+import com.kin.ecosystem.common.Observer
+import com.kin.ecosystem.common.exception.ClientException
+import com.kin.ecosystem.common.model.Balance
+import com.kin.ecosystem.core.Log
+import com.kin.ecosystem.core.Logger
+import com.kin.ecosystem.core.bi.EventLogger
+import com.kin.ecosystem.core.bi.RecoveryBackupEvents
+import com.kin.ecosystem.core.bi.RecoveryRestoreEvents
+import com.kin.ecosystem.core.bi.events.BackupWalletFailed
+import com.kin.ecosystem.core.data.blockchain.BlockchainSource
+import com.kin.ecosystem.core.data.settings.SettingsDataSource
+import com.kin.ecosystem.main.INavigator
+import com.kin.ecosystem.recovery.BackupAndRestoreCallback
+import com.kin.ecosystem.recovery.exception.BackupAndRestoreException
+import com.kin.ecosystem.settings.BackupManager
+import com.kin.ecosystem.settings.view.ISettingsView
+import com.kin.ecosystem.settings.view.ISettingsView.IconColor
+import com.kin.ecosystem.settings.view.ISettingsView.Item
+import java.math.BigDecimal
+
+class SettingsPresenter(private val settingsDataSource: SettingsDataSource,
+                        private val blockchainSource: BlockchainSource,
+                        private var backupManager: BackupManager,
+                        private val navigator: INavigator?,
+                        private val eventLogger: EventLogger) : BasePresenter<ISettingsView>(), ISettingsPresenter {
+
+    private var balanceObserver: Observer<Balance>? = null
+    private var currentBalance = blockchainSource.balance
+    private var publicAddress = blockchainSource.publicAddress
+
+    init {
+        registerToEvents()
+        registerToCallbacks()
+    }
+
+    override fun onResume() {
+        updateSettingsIcon()
+    }
+
+    override fun onPause() {
+        removeBalanceObserver()
+    }
+
+    override fun onDetach() {
+        super.onDetach()
+        backupManager.release()
+    }
+
+    private fun addBalanceObserver() {
+        balanceObserver = object : Observer<Balance>() {
+            override fun onChanged(value: Balance) {
+                currentBalance = value
+                if (isGreaterThenZero(value)) {
+                    updateSettingsIcon()
+                }
+            }
+        }.also {
```

Comment:
isn't it more short without the `also`, or am i missing something?

Reply:
more readable like that from my point of view.